stage_tester/ui_actions.py

```diff
@@ -0,0 +1,299 @@
+    """Entry point to this script."""
+    # Parse and process and command line arguments.
+    args = cli_arguments()
+
+    # check -c and -l args are not both provided
+    if args.cluster and args.cluster_list_file:
+        print(f"{sys.argv[0]}: error: Please provide cluster UUID through either -c or -l, not both.")
+        sys.exit(1)
+
+    # validate the recommendation to work with
+    selector = args.selector
+    """
+    Match any string that has alphanumeric characters separated by at least one dot
+    (".") before a vertical line ("|"), followed by only uppercase characters,
+    numbers, or underscores ("_")
+    """
+    if not re.match(r'[a-zA-Z_0-9]+\.[a-zA-Z_0-9.]+\|[A-Z_0-9]+$', selector):
+        print(f"{sys.argv[0]}: error: Please provide a valid rule selector (plugin|ek)")
+        sys.exit(1)
+
+    # validate operation(s) to execute
+    operations = args.operations[0]
+    print("operations:", operations)
+    for idx, op in enumerate(operations):
+        print("operation:", op)
+        if op not in ALLOWED_OPERATIONS:
+            # Only OK if it is the feedback for disable_with_feedback
+            if not (type(op) == str and operations[idx - 1] == "disable_with_feedback"):
+                print(f"{sys.argv[0]}: error: Received operation: {op}.")
+                print(f"{sys.argv[0]}: error: Expected one of {ALLOWED_OPERATIONS}.")
+                print(f"{sys.argv[0]}: error: Please provide a valid operation.")
+                sys.exit(1)
+        elif op == "disable_with_feedback":
+            if len(operations) == 1 or idx + 1 == len(operations) or operations[idx + 1] in ALLOWED_OPERATIONS:
+                # disable_with_feedback provided without feedback
+                print(f"{sys.argv[0]}: warning: {op} without feedback string")
+                register_operation("disable")
+            else:
+                register_operation(op, operations[idx + 1])
+        else:
+            register_operation(op)
+
+    verbose = args.verbose
+    proxies = {
+        'https': args.proxy
+    } if args.proxy else None
+    auth = (args.user, args.password)
+    clusters = {args.cluster} if args.cluster else set(open(args.cluster_list_file).read().split())
+    plugin = selector.split("|")[0]
+    error_key = selector.split("|")[1]
+
+    if verbose:
+        print("Proxy settings:", proxies)
+        print("Auth settings:", auth)
+        print("Address:", args.address)
+        print("Cluster(s):", clusters)
+        print("Rule selector:", selector)
+        print("Plugin:", plugin)
+        print("Error Key:", error_key)
+        print("Operations:", REGISTERED_OPERATIONS)
+
+    execute_operations(args.address, proxies, auth, clusters, plugin, error_key)
+
+
+def execute_operations(address, proxies, auth, clusters, plugin, error_key):
+    for cluster in clusters:
+        for item in REGISTERED_OPERATIONS:
+            print("working with", item)
+            op = item[0]
+            if op == "like":
```

Comment:
probably it would be better to use a dict of operations there, WDYT?

Reply:
Yeah, I can have a dict of key = operation, val = the corresponding request.
I thought it would complicate the code, but it doesn't seem like it =)

Reply:
I didn't use a dict because it complicates using multiple times the same operation. For example, "like dislike like" would end up with the "like" key registered only once. So the alternative was to have a list of operations, and I decided to go with a list of lists.

So it depends wether this kind of use case will be needed or not.

Reply:
no I meant to have an dict operation -> function, to avoid if-elseif chain

Reply:
done